src/dso_api/dynamic_api/filters/parser.py

```diff
@@ -440,25 +441,27 @@ def _to_orm_path(parts: list[FilterPathPart]) -> str:
     return "__".join(names)
 
 
-def _get_reverse_id_field(additional_relation: AdditionalRelationSchema) -> DatasetFieldSchema:
-    """Find the field that is targeted by direct queries against the reverse relation.
-    This is not the reverse field (a foreign key in the table), but the identifier field.
-    """
-    table = additional_relation.related_table
-    return table.get_field_by_id(table.identifier[0])
-
-
 class FilterPathPart(NamedTuple):
     """An entry in the filter path, translated to a field.
     The field and reverse field are so different in their API,
     that those fields are mentioned separately. They don't have a useful base class.
     """
 
-    name: str
+    id: str
     field: DatasetFieldSchema
     reverse_field: AdditionalRelationSchema | None = None
     is_many: bool = False
 
+    @property
+    def name(self):
+        # Simulates the same properties as the DatasetFieldSchema provides.
+        return toCamelCase(self.id)
+
+    @property
+    def python_name(self):
+        # Likewise, an exact replica of what DatasetFieldSchema provides.
+        return to_snake_case(self.id)
```

Comment:
If we put the casing logic on `AdditionalRelationSchema` as well, we can do:

```suggestion
    @property
    def name(self):
        return getattr(self.reverse_field or self.field, "name")

    @property
    def python_name(self):
        return getattr(self.reverse_field or self.field, "python_name")
```

That's -1 import of these functions.

Reply:
thanks, great fix! I was already puzzling with that one...